packages/video_player/video_player/test/video_player_test.dart

```diff
@@ -698,6 +723,7 @@ void main() {
         expect(controller.value.isPlaying, isTrue);
 
         await controller.pause();
+        await tester.runAsync(controller.dispose);
```

Comment:
For those ones, I didn't manage to make the test pass without using `runAsync`.

If I use
```dart
addTearDown(() {
  controller.dispose();
});
```

or 

```dart
// end of test
unawaited(controller.dispose());
// a lot of 
tester.pump();
tester.pump();
```

it doesn't fix the memory leak.

If I use 

```dart
addTearDown(controller.dispose());
```

or

```dart
// end of test
await controller.dispose();
```

the test hangs and times out.


Do you have an idea on how to do it better?

Reply:
It looks smelly for me that `dispose` is async. But, it is test only class, so it is ok. 
And thus `runAsync` makes perfect sense.

Reply:
Are these tests doing anything to test the real implementation of this class (VideoPlayerController)? In the actual usage of `dispose` there are asynchronous happenings. I'm not super familiar with how the leak testing works.

Reply:
From what I remember, they do. Those controller were created and I had issues making the tests wait for the dispose methods to finish